File: app/src/main/java/com/example/builday365/viewmodel/MapsViewModel.java
package com.example.builday365.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.google.android.gms.maps.model.LatLng;

import android.util.Log;

public class MapsViewModel extends ViewModel {
    private static final String TAG = "MapsViewModel";

    private MutableLiveData<Boolean> mFocused = new MutableLiveData<Boolean>();
    private MutableLiveData<LatLng> mLastLocation = new MutableLiveData<>();
    private MutableLiveData<LatLng> mHomeLocation = new MutableLiveData<>();

    public void setFocused(Boolean isFocused) {
        mFocused.setValue(isFocused);
    }

    public void setLastLocation(LatLng lastLocation) {
        this.mLastLocation.setValue(lastLocation);
    }

    public void setHomeLocation(LatLng homeLocation) {
        this.mHomeLocation.setValue(homeLocation);
    }

    public LiveData<Boolean> getFocused() {
        return mFocused;
    }

    public LiveData<LatLng> getLastLocation() {
        return mLastLocation;
    }

    public LiveData<LatLng> getHomeLocation() {
        return mHomeLocation;
    }

    public void onFabClick() {
        mFocused.setValue(true);
    }
}

File: app/src/main/java/com/example/builday365/viewmodel/MapsViewModel.kt
package com.example.builday365.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.google.android.gms.maps.model.LatLng

class MapsViewModel : ViewModel() {
    private val mFocused = MutableLiveData<Boolean>()
    private val mLastLocation = MutableLiveData<LatLng>()
    private val mHomeLocation = MutableLiveData<LatLng>()
    fun setFocused(isFocused: Boolean) {
        mFocused.value = isFocused
    }

    fun setLastLocation(lastLocation: LatLng) {
        mLastLocation.value = lastLocation
    }

    fun setHomeLocation(homeLocation: LatLng) {
        mHomeLocation.value = homeLocation
    }

    val focused: LiveData<Boolean>
        get() = mFocused
    val lastLocation: LiveData<LatLng>
        get() = mLastLocation
    val homeLocation: LiveData<LatLng>
        get() = mHomeLocation

    fun onFabClick() {
        mFocused.value = true
    }

    companion object {
        private const val TAG = "MapsViewModel"
    }
}